client/client.go

```diff
@@ -0,0 +1,64 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"istio.io/manager/model"
+)
+
+type ManagerClient struct {
+	base             *url.URL
+	versionedAPIPath string
+	client           *http.Client
+}
+
+func NewManagerClient(base *url.URL, apiVersion string, client *http.Client) *ManagerClient {
```

Comment:
This function is modifying the original `url.URL` pass despite the use of `trimmedBase`. You can pass by value instead of pointer if that is not intended.

Also, this could be rewritten to use strings.TrimSuffix and strings.TrimPrefix, e.g.

```golang
base.Path = strings.TrimSuffix(base.Path, "/")
apiVersion = strings.TrimPrefix(strings.TrimSuffix(apiVersion, "/"), "/")
```